### ISSUE TITLE
Naming scheme of logfiles creates racecondition when running many Xtrapolate instances in parallel

### ISSUE THREAD

**Issue by HatPdotS** [ElkeDeZitter/Xtrapol8#46]
Hello, 

I am PhD student in the Standfuss group at the Paul Scherrer Institut working on Time-resolved Crystallography. 
I am using Xtrapolate to create many (~400) q-weighted difference maps. The process is quite fast but I get random crashes of Xtrapolate. I believe this due to the naming scheme of the logfiles:

Traceback (most recent call last):
  File "/opt/psi/MX/Xtrapol8/20221122/Xtrapol8/Fextr.py", line 3125, in <module>
    run(sys.argv[1:])
  File "/opt/psi/MX/Xtrapol8/20221122/Xtrapol8/Fextr.py", line 2157, in run
    shutil.move(full_log, full_log.replace(log_dir,outdir))
  File "/afs/psi.ch/sys/psi.ra/MX/phenix/phenix-1.19.1-4122/conda_base/lib/python2.7/shutil.py", line 316, in move
    copy2(src, real_dst)
  File "/afs/psi.ch/sys/psi.ra/MX/phenix/phenix-1.19.1-4122/conda_base/lib/python2.7/shutil.py", line 144, in copy2
    copyfile(src, dst)
  File "/afs/psi.ch/sys/psi.ra/MX/phenix/phenix-1.19.1-4122/conda_base/lib/python2.7/shutil.py", line 96, in copyfile
    with open(src, 'rb') as fsrc:
IOError: [Errno 2] No such file or directory: '/das/work/units/LBR-FEL/p17490/Peter/2024-01-25_12h12_Xtrapol8.log'

I believe the error occurs when two Xtrapolate processes run at the same time and get the same name for log. Then only one log is written and moved. This leads to a crash in the second process as the log is not there to be moved. 
The solution would be to have unique log names I can specify through the Phil or writing the log directly into the output folder. 
A very simple workaround would be putting the shutil.move in a try except statement.

Best regards, 
Hans Peter Seidel

**Comment by ElkeDeZitter** [comment]
Dear Hans Peter Seidel (@HatPdotS),
Thanks for reporting this issue. 
Indeed, the problem is that Xtrapol8 log file contains the date and time up to min, which is not unique if multiple instances are rapidly launched (e.g. from a script). We have also encountered this problem in the past and made a custom fix. We will add it soon (together with some other small fixes), at latest by the end of next week.
I'll keep you posted!
Elke

**Comment by ElkeDeZitter** [comment]
Dear Hans Peter Seidel (@HatPdotS),

Sorry for the delay. The issue you were facing should have been solved with in latest Xtrapol8 version (v.1.2.3).

Note that when you run many jobs quickly one after the other, you should change the output directory. Otherwise, you will face the same issue as the one that you had before (I can give more information if you wish). We attempted to solve that issue too, but due to a lack of time and some further complication we decided to postpone.

Your Github issue is closed since it was linked to the developer branch that we just merged into the stable version.
Please let me know if you encounter further issues.

Elke
